utils/node_local_test.py

```diff
@@ -1300,6 +1312,163 @@ def run_in_fg(server, conf):
+            print('Creating {} dirs took {:.2f}'.format(count,
+                                                        elapsed))
+            os.rename(dir_dir, t_dir)
+
+        if not os.path.exists(t_file):
+            try:
+                os.mkdir(file_dir)
+            except FileExistsError:
+                pass
+            start = time.time()
+            for i in range(count):
+                f = open(os.path.join(file_dir, str(i)), 'w')
+                f.close()
+            elapsed = time.time() - start
+            print('Creating {} files took {:.2f}'.format(i + 1,
+                                                         elapsed))
+            os.rename(file_dir, t_file)
+
+        return time.time() - start_all
+
+    def print_results(results):
+        """Display the results"""
+        for count in results:
+            line = '{:-10d} '.format(count)
+            rd = results[count]
+            for name, value in rd.items():
+                line += '{} {:.2f} '.format(name, value)
+            print(line)
+
+    pools = get_pool_list()
+
+    while len(pools) < 1:
+        pools = make_pool(server)
+
+    pool = pools[0]
+
+    container = str(uuid.uuid4())
+
+    dfuse = DFuse(server, conf, pool=pool)
+
+    print('Creating container and populating')
+    count = 4
+    dfuse.start()
+    parent = os.path.join(dfuse.dir, container)
+    try:
+        os.mkdir(parent)
+    except FileExistsError:
+        pass
+    create_time = make_dirs(parent, count)
+    dfuse.stop()
+
+    results = OrderedDict()
+    results[count] = OrderedDict()
+    results[count]['create_time'] = create_time
+    all_start = time.time()
+
+    while True:
+
+        dfuse = DFuse(server, conf, pool=pool, container=container)
+        dir_dir = os.path.join(dfuse.dir,
+                               'dirs.{}'.format(count))
+        file_dir = os.path.join(dfuse.dir,
+                                'files.{}'.format(count))
+        dfuse.start()
+        start = time.time()
+        subprocess.run(['/bin/ls', '-U', dir_dir], stdout=subprocess.PIPE)
+        elapsed = time.time() - start
+        print('processed {} dirs in {:.2f} seconds'.format(count,
+                                                           elapsed))
+        results[count]['dirs'] = elapsed
+        dfuse.stop()
+        dfuse = DFuse(server, conf, pool=pool, container=container)
+        dfuse.start()
+        start = time.time()
+        subprocess.run(['/bin/ls', '-U', file_dir], stdout=subprocess.PIPE)
+        elapsed = time.time() - start
+        print('processed {} files in {:.2f} seconds'.format(count,
+                                                            elapsed))
+        results[count]['files'] = elapsed
+        dfuse.stop()
+
+        dfuse = DFuse(server, conf, pool=pool, container=container)
+        dfuse.start()
+        start = time.time()
+        subprocess.run(['/bin/ls', '-t', dir_dir], stdout=subprocess.PIPE)
```

Comment:
Not very familiar with ls and under what exact options it will perform stat(). does it make sense here to also try to avoid having ls do any sorting, while still having ls perform a stat on the entries within dir_dir? For example would /bin/ls -lU serve that purpose?

Reply:
It's tricky and I can comment it better.  ls -t will sort on time so thereby causing it to do a stat, ls -l will trigger it to do two getxattr calls as well so slow things down considerably.